src/Utils/request/api-request.ts

```diff
@@ -0,0 +1,60 @@
+import careConfig from "@careConfig";
+
+import { Route } from "@/Utils/request/types";
+import { makeHeaders, makeUrl } from "@/Utils/request/utils";
+
+import { getResponseBody } from "./request";
+
+interface QueryOptions {
+  pathParams?: Record<string, string>;
+  queryParams?: Record<string, string>;
+  silent?: boolean;
+}
+
+async function queryRequest<TData, TBody>(
+  { path, method, noAuth }: Route<TData, TBody>,
+  options?: QueryOptions & { signal?: AbortSignal },
+): Promise<TData> {
+  const url = `${careConfig.apiUrl}${makeUrl(path, options?.queryParams, options?.pathParams)}`;
+
+  const requestOptions: RequestInit = {
+    method,
+    headers: makeHeaders(noAuth ?? false),
+    signal: options?.signal,
+  };
+
+  const res = await fetch(url, requestOptions);
+
+  if (!res.ok) {
+    const error = await res
+      .json()
+      .catch(() => ({ detail: "Something went wrong!" }));
+    throw {
+      ...error,
+      code: error.code || (res.status === 404 ? "not_found" : undefined),
+      status: res.status,
+      silent: options?.silent,
+      detail: error.detail || "Something went wrong!",
+    };
```

Comment:
Let's stick with passing the object as `cause` in the Error? Since TanStack query's error callback uses the `Error` type. That way we won't need to cast to `any` type to hack around the type.

```suggestion
    throw new Error('Request Failed', {
      cause: {
        ...error,
        code: error.code || (res.status === 404 ? "not_found" : undefined),
        status: res.status,
        silent: options?.silent,
        detail: error.detail || "Something went wrong!",
    }
    });
```

We could even expand the Error class to suit for our needs and provide type-safety for them. See example here: https://dev.to/expressots/handling-custom-error-responses-from-expressots-with-tanstack-query-and-nextjs-1lij